imports/plugins/core/versions/server/util/convert68.js

```diff
@@ -0,0 +1,296 @@
+    shopId,
+    canSellWithoutInventory
+  });
+
+  // Now it's time to calculate top-level variant aggregated inventory and add those to the results.
+  // For non-sellable (parent) variants, we need to get inventory for all of their options
+  // and calculate aggregated values from them.
+  let childOptionResults = [];
+  if (parentVariantProductConfigurations.length) {
+    const variantIds = parentVariantProductConfigurations.map(({ productVariantId }) => productVariantId);
+    const allOptions = await Products.find({
+      ancestors: { $in: variantIds }
+    }, {
+      projection: {
+        ancestors: 1
+      }
+    }).toArray();
+
+    childOptionResults = await getInventoryResults(collections, {
+      fields,
+      productConfigurations: allOptions.map((option) => ({
+        productId: option.ancestors[0],
+        productVariantId: option._id
+      })),
+      shopId,
+      canSellWithoutInventory
+    });
+
+    for (const productConfiguration of parentVariantProductConfigurations) {
+      const childOptions = allOptions.filter((option) => option.ancestors.includes(productConfiguration.productVariantId));
+      const childOptionsInventory = childOptions.reduce((list, option) => {
+        const optionResult = childOptionResults.find((result) => result.productConfiguration.productVariantId === option._id);
+        if (optionResult) list.push(optionResult.inventoryInfo);
+        return list;
+      }, []);
+      results.push({
+        productConfiguration,
+        inventoryInfo: {
+          canBackorder: childOptionsInventory.some((option) => option.canBackorder),
+          inventoryAvailableToSell: childOptionsInventory.reduce((sum, option) => sum + option.inventoryAvailableToSell, 0),
+          inventoryInStock: childOptionsInventory.reduce((sum, option) => sum + option.inventoryInStock, 0),
+          inventoryReserved: childOptionsInventory.reduce((sum, option) => sum + option.inventoryReserved, 0),
+          isBackorder: childOptionsInventory.every((option) => option.isBackorder),
+          isLowQuantity: childOptionsInventory.some((option) => option.isLowQuantity),
+          isSoldOut: childOptionsInventory.every((option) => option.isSoldOut)
+        }
+      });
+    }
+  }
+
+  return results;
+}
+
+/**
+ * @param {Object} catalogProduct The catalog item to transform
+ * @param {Object} collections The catalog item to transform
+ * @returns {Object} The converted item document
+ */
+export async function convertCatalogItemVariants(catalogProduct, collections) {
+  const { AppSettings, Products } = collections;
+
+  const shopSettings = await AppSettings.findOne({ shopId: catalogProduct.shopId });
+
+  // By default, products that do not have inventory management enabled are sellable.
+  let canSellWithoutInventory = true;
+  if (shopSettings && typeof shopSettings.canSellVariantWithoutInventory === "boolean") {
+    canSellWithoutInventory = shopSettings.canSellVariantWithoutInventory;
+  }
```

Comment:
Could `_.get` here to compress these four lines into one.

Reply:
code consolidated